tests/Client/UpdaterTest.php

```diff
@@ -1317,4 +1324,62 @@ public function providerAttackRepoException(): array
             ],
         ];
     }
+
+    public function providerKeyRotation(): array
+    {
+        return [
+            'not rotated' => [
+                'PublishedTwice',
+                [
+                    'root' => 2,
+                    'timestamp' => 1,
+                    'snapshot' => 1,
+                    'targets' => 1,
+                ],
+            ],
+            'timestamp rotated' => [
+                'PublishedTwiceWithRotatedKeys_timestamp',
+                [
+                    'root' => 2,
+                    'timestamp' => null,
+                    'snapshot' => null,
+                    'targets' => 1,
+                ],
+            ],
+            'snapshot rotated' => [
+                'PublishedTwiceWithRotatedKeys_snapshot',
+                [
+                    'root' => 2,
+                    'timestamp' => null,
+                    'snapshot' => null,
+                    'targets' => 1,
+                ],
+            ],
+        ];
+    }
+
+    /**
+     * Tests that the updater correctly handles key rotation.
```

Comment:
either in the test provider cases or the comment here we should mention that we expect the timestamp and snapshot files to be delete if *either* keys are rotated.

Also we should reference the part in the spec that we are covering as is now in main after #248
Also we could add `@covers`'s  for `\Tuf\Client\Updater::updateRoot()` and `\Tuf\Client\Updater::hasRotatedKeys()`

Reply:
Done!